addons/payment_payulatam/models/payment.py

```diff
@@ -37,8 +38,9 @@ def _payulatam_generate_sign(self, inout, values):
             data_string = ('~').join((self.payulatam_api_key, self.payulatam_merchant_id, values['referenceCode'],
                                       str(values['amount']), values['currency']))
         else:
+            rounded_amount = decimal.Decimal(values.get('TX_VALUE')).quantize(decimal.Decimal('0.1'), decimal.ROUND_HALF_EVEN)
```

Comment:
```suggestion
            rounded_amount = decimal.Decimal(values.get('TX_VALUE')).quantize(values['currency'].rounding, decimal.ROUND_HALF_EVEN)
```

Reply:
Not sure if values['currency'] is a browse record @backspac

Reply:
As you said @backspac not important as it is only for the signing.  and the error returned said approximation to a decimal